src/math/random.js

```diff
@@ -144,22 +160,37 @@ p5.prototype.random = function(min, max) {
   } else {
     rand = Math.random();
   }
+
   if (typeof min === 'undefined') {
     return rand;
-  } else if (typeof max === 'undefined') {
-    if (min instanceof Array) {
-      return min[Math.floor(rand * min.length)];
+  } else if (min instanceof Array) {
```

Comment:
Can we change this check to use `Array.isArray`, `instanceof` checks may cause problems in rare cases.

Reply:
Sure; changed it.